src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -60,10 +62,21 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
+            // TODO: the line below once .NET ships new Activity
+            // Or do reflection now.
+            // activity.ActivityKind = ActivityKind.Server
+
             var request = context.Request;
 
             // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
             var path = (request.PathBase.HasValue || request.Path.HasValue) ? (request.PathBase + request.Path).ToString() : "/";
+            if (activity != null)
+            {
+                activity.DisplayName = path;
+            }
+
+            var samplingDecision = this.sampler.ShouldSample(activity.Context, activity.TraceId, default, path, activity.Kind, activity.Tags, activity.Links);
```

Comment:
This is the part which makes me not 100% convinved this approach is right. All instrumentationadatpers now need to know about Samplers (which is SDK concept).

Instead of simply checking if activity.IsAllDataRequested, its now first setting IsAllDataRequested by running sampling algorithm